RockLib.HealthChecks.WebApi/HealthCheckHandler.cs

```diff
@@ -5,41 +5,40 @@
 using System.Threading;
 using System.Threading.Tasks;
 
-namespace RockLib.HealthChecks.WebApi
+namespace RockLib.HealthChecks.WebApi;
+
+/// <summary>
+/// A message handler for running health checks.
+/// </summary>
+public sealed class HealthCheckHandler : HttpMessageHandler
 {
+    private readonly IHealthCheckRunner _healthCheckRunner;
+    private readonly bool _indent;
+
     /// <summary>
-    /// A message handler for running health checks.
+    /// Initializes a new instance of the <see cref="HealthCheckHandler"/> class.
     /// </summary>
-    public sealed class HealthCheckHandler : HttpMessageHandler
+    /// <param name="healthCheckRunner">
+    /// The <see cref="IHealthCheckRunner"/> that evaluates the health of the service.
+    /// </param>
+    /// <param name="indent">Whether to indent the JSON output.</param>
+    public HealthCheckHandler(IHealthCheckRunner healthCheckRunner, bool indent)
     {
-        private readonly IHealthCheckRunner _healthCheckRunner;
-        private readonly bool _indent;
+        _healthCheckRunner = healthCheckRunner ?? throw new ArgumentNullException(nameof(healthCheckRunner));
```

Comment:
why aren't you using the NET6_0_OR_GREATER conditional and ThrowIfNull here as well? Are we aiming to make the ThrowIfNull (for >= .NET 6) a RockLib standard going forward?

Reply:
Good question. I have to admit, I like the approach above, especially when you're just assigning to a field or property. I just forget that `??` exists.

Reply:
Plus, that's not .NET6+ specific.

I would say, if you just need to check that a parameter is not null and you're not assigning it, use `ThrowIfNull()`. Otherwise, do something like what's done here.